src/rgw/rgw_client_io_filters.h

```diff
@@ -182,8 +226,16 @@ size_t BufferingFilter<T>::complete_request()
   size_t sent = 0;
 
   if (! has_content_length) {
+    /* It is not correct to count these bytes here,
+     * because they can only be part of the header.
+     * Therefore force count to 0.
+     */
     sent += DecoratedRestfulClient<T>::send_content_length(data.length());
     sent += DecoratedRestfulClient<T>::complete_header();
+    ldout(cx, 30) <<
+        "BufferingFilter::complete_request: !has_content_length: IGNORE: sent="
+        << sent << dendl;
+    sent = 0;
```

Comment:
I'm researching this possibility. The idea is to simplify the accounting in a way that takes care only of the existing use cases (HTTP Body accounting), and thus allows to strip the extra complexity we have at the moment.

I think we still should merge the patch as a stop-gap solution.